src/main/java/ch/njol/skript/classes/data/JavaClasses.java

```diff
@@ -74,8 +82,8 @@ public class JavaClasses {
 	 * </p>
 	 */
 	public static final Pattern DECIMAL_PATTERN =
-		Pattern.compile("(?<num>" + DECIMAL_NUMBER_PATTERN + ")" +
-			"(?: (?:in )?(?:(?<rad>rad(?:ian)?s?)|deg(?:ree)?s?))?");
+		Pattern.compile("(?<num>%s%s)(?: (?:in )?(?:(?<rad>rad(?:ian)?s?)|deg(?:ree)?s?))?"
+			.formatted(DECIMAL_NUMBER_PATTERN, SCIENTIFIC_PATTERN));
```

Comment:
not required but maybe a good chance to reduce some of the duplication in the radian/degree part of the pattern too